packages/x-date-pickers/src/DatePicker/shared.ts

```diff
@@ -36,7 +36,7 @@ export interface BaseDatePickerSlotsComponentsProps<TDate>
 }
 
 export interface BaseDatePickerProps<TDate>
-  extends ExportedDateCalendarProps<TDate>,
+  extends Omit<ExportedDateCalendarProps<TDate>, 'value' | 'onChange' | 'defaultValue'>,
```

Comment:
The calendars have the same `value` / `onChange` / `defaultValue` as the new pickers
But for the legacy pickers, I have to omit here and re-define to avoid type conflicts